azkaban-exec-server/src/main/java/azkaban/execapp/reporter/AzkabanEventReporter.java

```diff
@@ -0,0 +1,20 @@
+package azkaban.execapp.reporter;
+
+import azkaban.event.Event;
+import java.util.Map;
+
+/**
+ * Implement this interface to report flow and job events. Event reporter
+ * can be turned on by setting the property 'event.reporter.enabled' to true.
+ *
+ * By default, a KafkaAvroEventReporter is provided. Alternate implementations
+ * can be provided by setting the property 'event.reporter.class'.
+ *
+ * The constructor will be called with a azkaban.utils.Props object passed as
+ * the only parameter. If such a constructor doesn't exist, then the AzkabanEventReporter
+ * instantiation will fail.
+ */
+public interface AzkabanEventReporter {
+
+  boolean report(Event event, Map<String, String> metadata);
```

Comment:
Event class is an implementation detail and it includes methods you don't need for this interface. Consider defining an event interface for this purpose?

We have an spi ( service provider interface ) module. e.g. azkaban.spi.Storage
Should this API go into that module? Clients can depend on this module to get the API definitions.